gotocompany/shield/v1beta1/shield.proto

```diff
@@ -1109,14 +1109,27 @@ message UpdateResourceResponse {
   Resource resource = 1;
 }
 
-message CheckResourcePermissionRequest {
+message ResourcePermission {
   string object_id = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
 
   string object_namespace = 2 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
 
   string permission = 3 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$"];
 }
 
-message CheckResourcePermissionResponse {
-  bool status = 1;
+message CheckResourcePermissionRequest {
+  string object_id = 1 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$",deprecated = true];
+  string object_namespace = 2 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$",deprecated = true];
+  string permission = 3 [(validate.rules).string.pattern = "^[A-Za-z0-9_-]+$",deprecated = true];
+
+  repeated ResourcePermission resource_permissions = 4;
 }
+
+message CheckResourcePermissionResponse {
+  message ResourcePermissionResponse {
+    ResourcePermission resource_permission = 1;
+    bool allowed = 2;
+  }
+  bool status = 1 [deprecated = true];
+  repeated ResourcePermissionResponse resource_permissions = 2;
```

Comment:
can you share example for this? i believe this would ResourcePermission would be nested and not on same root

Reply:
that's right @krtkvrm .

CheckResourcePermissionRequest 
`{
"resource_permissions": [
 {"object_id","" ...},  {"object_id","" ...}
]
}`

CheckResourcePermissionResponse 
`{
"resource_permissions": [
 {
 "resource_permission: {"object_id":".."},
"allowed": true,
},
 {
 "resource_permission: {"object_id":".."},
"allowed": true,
}
]
}`

We can avoid the nested 'resource_permission' field by defining the  object_id, object_namespace, and permission inside the `ResourcePermissionResponse`.  it felt like a duplication. WDYT?

Reply:
@krtkvrm, updated the `CheckResourcePermissionResponse` to match the proposed contract.